back/engines/commercial/analysis/app/lib/analysis/summarization_method/base.rb

```diff
@@ -4,37 +4,49 @@ module Analysis
   class SummarizationMethod::Base
     attr_reader :analysis, :task, :summary, :input_to_text
 
+    SUMMARIZATION_METHOD_CLASSES = [
+      SummarizationMethod::OnePassLLM
+    ]
+
+    LLMS = [
+      LLM::GPT48k.new,
+      LLM::GPT432k.new,
+      LLM::GPT3516k.new
+    ]
+
     class SummarizationFailedError < StandardError; end
+    class TooManyInputs < SummarizationFailedError; end
 
-    def self.for_summarization_method summarization_method, *params, **kwargs
-      case summarization_method
-      when 'gpt4'
-        SummarizationMethod::Gpt4.new(*params, **kwargs)
-      when 'bogus'
-        SummarizationMethod::Bogus.new(*params, **kwargs)
-      else
-        raise ArgumentError, "Unsupported summarization_method #{summarization_method}"
-      end
+    def self.plan(summary)
+      SummarizationMethod::OnePassLLM.new(summary).generate_plan || SummarizationPlan.new(
```

Comment:
It's true that `Base` now has knowledge about `OnePassLLM`. I've been going in all directions with this implementation, but in the end we won't be able to avoid this when we support more methods (which we very likely will). This `self.plan` method will have to return the optimal method to use, and to do so required internal knowledge of how and how well these methods work.

It is true that this specific error could be triggered by the method, but as in the general case the abstraction will be leaky, I figured this to be acceptable too. I'll keep it like this if you're fine with it, I will give this some more thought once we add extra methods.

Reply:
Sure, not a blocker.